src/main/java/Application/OpenCVServer/OpenCVServerOperator.java

```diff
@@ -0,0 +1,164 @@
+package Application.OpenCVServer;
+
+import java.io.*;
+import java.util.Arrays;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Vector;
+import java.net.*;
+
+import javax.imageio.ImageIO;
+import java.awt.image.BufferedImage;
+
+import Application.Utilities.DetectMarker;
+import Application.Utilities.HeartBeatRunner;
+import Application.Utilities.ImageProcessor;
+import Application.Utilities.OpenCVUtil;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import org.opencv.core.Mat;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+
+public class OpenCVServerOperator {
+    private ServerSocket serverSocket;
+    private Socket clientSocket;
+    private DataInputStream in;
+    private PrintWriter out;
+    private DetectMarker detector;
+    private int port;
+
+    private String fileName;
+
+    private BufferedImage currentImage;
+
+    public ServerRunner serverRunner;
+    public HeartBeatRunner beatRunner;
+
+    private static OpenCVServerOperator instance = null;
+
+    public OpenCVServerOperator() {
+        fileName= "read.png";
+    }
+
+    public void startConnection() throws IOException {
+        serverSocket = new ServerSocket(port);
+        clientSocket = serverSocket.accept();
+        out = new PrintWriter(clientSocket.getOutputStream(), true);
+        in = new DataInputStream(clientSocket.getInputStream());
+    }
+
+    // change name to something more appropriate
+    public void standbyForConnection() throws InvalidObjectException, IllegalArgumentException, JsonProcessingException, IOException {
+        //if input is faulty, timeout and continue
+        currentImage = ImageIO.read(ImageIO.createImageInputStream(clientSocket.getInputStream()));
+        ImageIO.write(currentImage, "png", new File(fileName));
+        Mat subject = ImageProcessor.getImageMat(fileName);
+
+        detector = new DetectMarker();
+        detector.detect(subject);
+        String ids = OpenCVUtil.serializeMat(detector.getIds());
+        String corners = OpenCVUtil.serializeMatList(detector.getCorners());
+        String result = "";
+
+        ObjectMapper mapper = new ObjectMapper();
+        ObjectNode resultObject = mapper.createObjectNode();
+        resultObject.put("ids", ids);
+        resultObject.put("corners", corners);
+        System.out.println(ids);
+        System.out.println(corners);
+
+        result = mapper.writeValueAsString(resultObject);
+
+        out.println(result);
+
+    }
+
+    public void stopConnection() throws IOException {
+        in.close();
+        out.close();
+        clientSocket.close();
+        serverSocket.close();
+    }
+
+    public static OpenCVServerOperator getInstance() {
+        if (instance == null) {
+            instance = new OpenCVServerOperator();
+        }
+        return instance;
+    }
+
+    public void setupServerRunners(String args[]) throws IllegalArgumentException {
```

Comment:
Again declaration in C style
```suggestion
    public void setupServerRunners(String[] args) throws IllegalArgumentException {
```